src/PayID/single-network-pay-id-client.ts

```diff
@@ -0,0 +1,341 @@
+   * - xrpl-testnet
+   * - xrpl-mainnet
+   * - eth-rinkeby
+   * - ach
+   *
+   * TODO(keefertaylor): Link a canonical list at payid.org when available.
+   *
+   * @param network The network that addresses will be resolved on.
+   * @param useHttps Whether to cuse HTTPS when making PayID requests. Most users should set this to 'true' to avoid
+   *                 Man-in-the-Middle attacks. Exposed as an option for testing purposes. Defaults to true.
+   */
+  constructor(
+    public readonly network: string,
+    private readonly useHttps: boolean = true,
+  ) {}
+
+  /**
+   * Retrieve the address associated with a PayID.
+   *
+   * @param payId The payID to resolve for an address.
+   * @returns An address representing the given PayID.
+   */
+  async addressForPayId(payId: string): Promise<CryptoAddressDetails> {
+    const payIdComponents = SingleNetworkPayIdClient.parsePayId(payId)
+    const basePath = this.useHttps
+      ? `https://${payIdComponents.host}`
+      : `http://${payIdComponents.host}`
+
+    // Swagger API adds the leading '/' in path automatically because it is part of the endpoint.
+    const path = payIdComponents.path.substring(1)
+
+    const api = new DefaultApi(undefined, basePath)
+
+    const options = SingleNetworkPayIdClient.makeOptionsWithAcceptTypes(
+      `application/${this.network}+json`,
+    )
+
+    try {
+      const { data } = await api.resolvePayID(path, options)
+      // TODO(keefertaylor): make sure the header matches the request.
+      if (data?.addresses) {
+        // With a specific network, exactly one address should be returned by a PayID lookup.
+        if (data.addresses.length === 1) {
+          return data.addresses[0].addressDetails
+        } else {
+          return Promise.reject(
+            new PayIDError(
+              PayIDErrorType.UnexpectedResponse,
+              'Received more addresses than expected',
+            ),
+          )
+        }
+      } else {
+        return Promise.reject(
+          new PayIDError(
+            PayIDErrorType.UnexpectedResponse,
+            'Too many addresses returned',
+          ),
+        )
+      }
+    } catch (error) {
+      if (error.response?.status === 404) {
+        const message = `Could not resolve ${payId} on network ${this.network}`
+        return Promise.reject(
+          new PayIdError(PayIdErrorType.MappingNotFound, message),
+        )
+      } else {
+        const message = SingleNetworkPayIdClient.messageFromMaybeHttpError(
+          error,
+        )
+        return Promise.reject(
+          new PayIdError(PayIdErrorType.UnexpectedResponse, message),
+        )
+      }
+    }
+  }
+
+  /**
+   * Generate a new invoice with compliance requests.
+   *
+   * @param payId The Pay ID to request an invoice for.
+   * @param nonce A randomly selected nonce that is unique to this invoice request.
```

Comment:
PayID no longer accepts a nonce from the client, the server returns a `txId` instead. This is because we were essentially giving the client authority to determine the identifier for something the server would need to store. 

On top of that this all might get tossed out the window in favor of the TRISA stuff lol